sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowEvaluatorFactoryBase.scala

```diff
@@ -101,6 +101,8 @@ trait WindowEvaluatorFactoryBase {
         val boundExpr = (expr.dataType, boundOffset.dataType) match {
           case (DateType, IntegerType) => DateAdd(expr, boundOffset)
           case (DateType, _: YearMonthIntervalType) => DateAddYMInterval(expr, boundOffset)
+          case (DateType, _: DayTimeIntervalType) =>
+            DateAdd(expr, ExtractANSIIntervalDays(boundOffset))
```

Comment:
I am not sure that discarding the time part of interval is appropriate approach.

Reply:
@mihailom-db BTW, what does the ANSI SQL standard says about that?

Reply:
Hm, true, I missed that this is might be start-end interval and not only start. Will rework it now. As for ANSI SQL I am not sure. @srielau do you have info on this?

Reply:
@MaxGekk just checked. This is the right approach I would say. We are not trying to change addition for Date + Interval here. We have analyser rule that converts add(date, interval) to this same expression as in my PR. We are just adding a case to properly handle case when we need to add interval to a date.

Reply:
Also, date does not have hours, mins and seconds fields.

Reply:
I'm missing context what is this about?

Reply:
I need to fix tests, but basically in windowFrame if we have DayTimeIntervalType for start_frame, and date as current column, we will fail instead of doing the window function properly.

Reply:
are you saying that Spark already supports `date + day_time_interval`? what's the behavior?

Reply:
we convert a date to a timestamp implicitly:
```sql
> select date'now'+interval '1 10:11:12.123456' day to second;
2024-11-20 10:11:12.123456
```

Reply:
Here @mihailom-db proposes to discard the time part at all, and get only days, for example above:
```sql
> select date'now'+interval '1 10:11:12.123456' day to second;
2024-11-20
```
Probably, it is ok in this case. @mihailom-db Could you add more tests at least for the cases:
```
    | frameType=RANGE BETWEEN start=frameBound AND end=frameBound
    | frameType=ROWS BETWEEN start=frameBound AND end=frameBound
```

Reply:
should we fail at runtime if the day-time interval contains non-day fields? We can relax it later (using truncation) if needed.

Reply:
ah good point, we have this information in the data type, we should only support INTERVAL DAY TO DAY